### PR TITLE
Check for Video Capture Device interference on Windows

### PR DESCRIPTION
**Pull request by prgmitchell**
### Description

This checks to see if a user has added multiple video capture devices pointing to the same device on Windows, then throws a warning if they do.

firefox_Mh6v67hzlU
Screenshot 2023-03-02 at 7 11 39 AM

### Motivation and Context

This comes up often enough in support and there's no clear indication to the user that they shouldn't be doing this.

### How Has This Been Tested?

Tested locally with this log with these two logs:
https://obsproject.com/logs/2DVdXkIVmhXGxVeR
https://obsproject.com/logs/mi4bNDc0I3xjjf4B

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: checks/windows.py [hunk 1]
```diff
@@ -350,3 +350,37 @@ def checkWindowsARM64EmulationStatus(lines):
...
+        for i in videoPaths:
+            if i not in pathList:
+                pathList.append(i)
+            else:
+                if (i != 'video path: '):
+                    duplicateList.append(i)
+
+        if (duplicateList):
+            return [LEVEL_WARNING, "Video Capture Device Interference",
+                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding sources to new scenes."]
```
Comment (RytoEX) [inline checks/windows.py:386]: Is this language pulled from the support bot or is it brand new?

Comment (prgmitchell) [reply checks/windows.py:386]: It is new, currently we have no checks or bot commands related to this.

**Comment by prgmitchell** [comment]
I have decided to update this PR to include a list of sources that would make it more clear to the user where the issue is.

File: checks/windows.py [hunk 2]
```diff
@@ -350,3 +351,50 @@ def checkWindowsARM64EmulationStatus(lines):
...
+                deviceName = deviceName[:-2]
+                deviceName = deviceName[2:]
+                if (dupListString == ''):
+                    dupListString = '<ul><li>' + deviceName
+                else:
+                    dupListString = dupListString + '</li><li>' + deviceName
+
+        if (duplicateList):
+            return [LEVEL_WARNING, "Video Capture Device Interference",
+                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding sources to new scenes.<br><br>The following source list contains duplicates:<br>" + dupListString]
```
Comment (RytoEX) [inline checks/windows.py:400]:
```suggestion
                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding the same Video Capture Device to new scenes.<br><br>The following source list contains duplicates:<br>" + dupListString]
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: checks/windows.py (hunk 2 above)
Comment (prgmitchell) [reply checks/windows.py]: PR updated with this wording.

**Comment by prgmitchell** [comment]
I have updated the PR for the wording changes requested. I have also updated the code to group duplicate devices if the user has 2 sources pointed at one device and 2 sources pointed at another for example. This makes it easier for a user to understand which sources are pointed at which device if they have multiple. I have used an <hr> border for this and matched the color with the log analyzer CSS.

**Comment by prgmitchell** [comment]
I have converted this to a draft as I found a bug in the original code.

**Comment by prgmitchell** [comment]
Having had some time to think about this, I am going to close this PR. Because a user can change their capture device settings and they are logged each time, most users logs will result in this flag if they are changing capture device settings trying to get it working.

**Comment by RytoEX** [comment]
Fair enough. I'd meant to follow-up and ask about how it handles multiple device settings changes, but this got lost in my to-do list.

**Pull request closed without merging**